r1184022 | rdale | 2010-10-09 08:51:34 +0200 (Sat, 09 Oct 2010) | 5 lines
Changed paths:
   M /trunk/KDE/kdebindings/generator/parser/rpp/pp-macro-expander.cpp
   M /trunk/KDE/kdebindings/generator/type_compiler.cpp

* Apply patch from Rex Dieter for fixing issues with parsing the
QtOpenGL module. Thanks for the patch.

Applied-Upstream: 4.6.0

Index: kdebindings/generator/type_compiler.cpp
===================================================================
--- kdebindings/generator/type_compiler.cpp	(revision 1184021)
+++ kdebindings/generator/type_compiler.cpp	(revision 1184022)
@@ -182,7 +182,7 @@
     TypeCompiler tc(m_session, m_visitor);
     tc.run(node->type_specifier, node->declarator);
     NameCompiler name_cc(m_session, m_visitor);
-    if (tc.type().isFunctionPointer())
+    if (tc.type().isFunctionPointer() && node->declarator && node->declarator->sub_declarator)
         name_cc.run(node->declarator->sub_declarator->id);
     else if (node->declarator)
         name_cc.run(node->declarator->id);
Index: kdebindings/generator/parser/rpp/pp-macro-expander.cpp
===================================================================
--- kdebindings/generator/parser/rpp/pp-macro-expander.cpp	(revision 1184021)
+++ kdebindings/generator/parser/rpp/pp-macro-expander.cpp	(revision 1184022)
@@ -268,10 +268,10 @@
         output << '\"';
 
         while (!is.atEnd()) {
-          if (input == '"') {
+          if (is == '"') {
             output << '\\' << is;
 
-          } else if (input == '\n') {
+          } else if (is == '\n') {
             output << '"' << is << '"';
 
           } else {
@@ -338,7 +338,7 @@
             
             skip_blanks(input, devnull());
             //Omit paste tokens behind empty used actuals, else we will merge with the previous text
-            if(input == '#' && (++input) == '#') {
+            if(!input.atEnd() && input == '#' && !(++input).atEnd() && input == '#') {
               ++input;
               //We have skipped a paste token
             }else{
